cmd/otel-allocator/prehook/relabel_test.go

```diff
@@ -0,0 +1,194 @@
+			// never exist and which ones will exist after relabelling, so these targets
+			// are kept in this step (will later be dealt with in Prometheus' relabel_config step)
+			cfg: relabel.Config{
+				SourceLabels: model.LabelNames{"label_not_present"},
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Action:       "keep",
+			},
+			isDrop: false,
+		},
+		{
+			cfg: relabel.Config{
+				SourceLabels: model.LabelNames{"i"},
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				Action:       "drop",
+			},
+			isDrop: true,
+		},
+		{
+			cfg: relabel.Config{
+				SourceLabels: model.LabelNames{"collector"},
+				Regex:        relabel.MustNewRegexp("(collector.*)"),
+				Action:       "drop",
+			},
+			isDrop: true,
+		},
+		{
+			cfg: relabel.Config{
+				SourceLabels: model.LabelNames{"i"},
+				Regex:        relabel.MustNewRegexp("bad.*match"),
+				Action:       "keep",
+			},
+			isDrop: true,
+		},
+		{
+			cfg: relabel.Config{
+				SourceLabels: model.LabelNames{"collector"},
+				Regex:        relabel.MustNewRegexp("collectors-n"),
+				Action:       "keep",
+			},
+			isDrop: true,
+		},
+	}
+
+	DefaultDropRelabelConfig = relabel.Config{
+		SourceLabels: model.LabelNames{"i"},
+		Regex:        relabel.MustNewRegexp("(.*)"),
+		Action:       "drop",
+	}
+)
+
+type RelabelConfigObj struct {
+	cfg    relabel.Config
+	isDrop bool
+}
+
+func colIndex(index, numCols int) int {
+	if numCols == 0 {
+		return -1
+	}
+	return index % numCols
+}
+
+func makeNNewTargets(n int, numCollectors int, startingIndex int) (map[string]*targetscommon.TargetItem, int, map[string]*targetscommon.TargetItem, map[string][]*relabel.Config) {
+	toReturn := map[string]*targetscommon.TargetItem{}
+	expectedMap := make(map[string]*targetscommon.TargetItem)
+	numItemsRemaining := n
+	relabelConfig := make(map[string][]*relabel.Config)
+	for i := startingIndex; i < n+startingIndex; i++ {
+		collector := fmt.Sprintf("collector-%d", colIndex(i, numCollectors))
+		label := model.LabelSet{
+			"collector": model.LabelValue(collector),
+			"i":         model.LabelValue(strconv.Itoa(i)),
+			"total":     model.LabelValue(strconv.Itoa(n + startingIndex)),
+		}
+		jobName := fmt.Sprintf("test-job-%d", i)
+		newTarget := targetscommon.NewTargetItem(jobName, "test-url", label, collector)
+		// add a single replace, drop, or keep action as relabel_config for targets
+		// index := rand.Intn(len(RelabelConfigs))
+		var index int
+		ind, _ := rand.Int(rand.Reader, big.NewInt(int64(len(RelabelConfigs))))
+
+		index = int(ind.Int64())
+
+		relabelConfig[jobName] = []*relabel.Config{
+			&RelabelConfigs[index].cfg,
+		}
+
+		targetKey := newTarget.Hash()
+		if RelabelConfigs[index].isDrop {
+			numItemsRemaining--
+		} else {
+			expectedMap[targetKey] = newTarget
+		}
+		toReturn[targetKey] = newTarget
+	}
+	return toReturn, numItemsRemaining, expectedMap, relabelConfig
+}
+
+func TestApply(t *testing.T) {
```

Comment:
Can you add a test for calling `Apply()` when the `relabelCfg` is empty? Are there any other edge cases we need to worry about?

Reply:
Added a followup test in `TestApply` which clears out the relabelCfg values. Also added `TestApplyEmpty` which will test when relabelCfg is an empty map. Thanks for the suggestion! I will continue to think about more cases I need to cover.